crates/providers/src/signer.rs

```diff
@@ -127,7 +126,7 @@ mod tests {
         // note: we need to 1) add <_, Ethereum> 2) layer things, and then 3) call .network before
         // we can call provider
         let provider = ProviderBuilder::<_, Ethereum>::new()
-            .layer(SignerLayer::new(EthereumSigner::from(wallet)))
+            .signer(EthereumSigner::from(wallet))
```

Comment:
would love if this could be `.signer(wallet)` via `impl From<Wallet> for EthereumSigner` and `impl From<Wallet> for SignerLayer` perhaps and making `fn signer<S: Into<SignerLayer>>` or something similar

Reply:
done